gridstatus/ieso.py

```diff
@@ -0,0 +1,517 @@
+NAMESPACES_FOR_XML = {"": "http://www.ieso.ca/schema"}
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
+
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get 5-minute load for the Market and Ontario for a given date or from
+        date to end date.
+
+        Args:
+            date (datetime.date | datetime.datetime | str): The date to get the load for
+                Can be a `datetime.date` or `datetime.datetime` object, or a string
+                with the values "today" or "latest". If `end` is None, returns
+                only data for this date.
+            end (datetime.date | datetime.datetime, optional): End date. Defaults None
+                If provided, returns data from `date` to `end` date. The `end` can be a
+                `datetime.date` or `datetime.datetime` object.
+            verbose (bool, optional): Print verbose output. Defaults to False.
+            frequency (str, optional): Frequency of data. Defaults to "5min".
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        # We have to add special time handling here because the existing utils
+        # don't perfectly handle the 1 hour data files.
+        if isinstance(date, tuple):
+            date, end = date
+
+        today = utils._handle_date("today", tz=self.default_timezone)
+
+        # Return data from the earliest interval today to the latest interval today
+        if date in ["today", "latest"]:
+            date = today
+            end = pd.Timestamp.now(tz=self.default_timezone)
+
+        # If given a date string or date set date to the earliest interval
+        # and end to the latest interval on the date
+        if isinstance(date, str) or (
+            isinstance(date, datetime.date) and not isinstance(date, datetime.datetime)
+        ):
+            date = utils._handle_date(date, tz=self.default_timezone)
+
+            if not end:
+                end = date + pd.Timedelta(days=1)
+
+        # Set end to the beginning of the next day to get the full day's worth of
+        # data on the end date.
+        if isinstance(end, str) or (
+            isinstance(end, datetime.date) and not isinstance(end, datetime.datetime)
+        ):
+            end = utils._handle_date(end, tz=self.default_timezone) + pd.Timedelta(
+                days=1,
+            )
+
+        if date.date() > today.date():
```

Comment:
i suppose there is no harm, but we dont need to include this check. we dont do it anywhere else if users request future dates.

Reply:
Without this check, the user will just see a series of 404s since the files don't exist. I think it's good to provide useful error messages to users.

Reply:
that makes sense, but i feel like it will get overwhelming to add this to every function. 

would probably make sense to generalize earliest/lastest available data into a decorator. 

that's out of scope for this PR, so we can leave this as if you want and circle back to a more generalized approach

Reply:
Could we add this logic to the `support_date_range` decorator? That seems a logical place for it, or we could add another decorator like `validate_dates`.

Reply:
ya, i think so that would make sense to add to existing decorator rather than create new one